cumulus_library_data_metrics/meta/base.jinja

```diff
@@ -0,0 +1,6 @@
+-- This is custom meta that isn't necessarily understood by Cumulus Library.
+-- Stuff that might be useful for downstream consumers.
+
+CREATE TABLE data_metrics__meta AS
+SELECT
+    date(from_iso8601_timestamp('{{ now }}')) AS build_date
```

Comment:
Matt: thoughts on this sort of thing?

I took the raw `meta` table name. And threw in this field. Is this field generally useful and something we should put in the library? Is this table name a good place for a study to throw study-specific fields? Or would you prefer something like `meta_custom` or something.

Reply:
So just recapping the as found state:
- Some studies have specific meta tables that are created (core has `meta_date`, `meta_version`) for singleton values related to some facet of the study's logic
- All studies get a `lib_transactions` table capturing their individual run times, including start and finish date.

This is not an intentional convention, and we could talk about it - but i do like the specificity in the different meta tables, which can be created at different points in a study's build and for different use cases. And I also sort of think that this one in particular is duplicative, partially, of what is going into the transactions table, though its buried a bit.

So from the hip, I think I would lean towards a more specific name, and/or having a view that look at the most recent finished item in the transactions table. But back to the 'every study is an island' approach we seem to be leaning towards, I don't feel strongly about it.

Reply:
#### Ah perfect
I forgot about `lib_transactions` - that's probably enough for this use case.

#### Study Metadata Namespacing
Well... you may like the specificity of table names, but studies need a safe namespace. Like let's say this field is not duplicative of `lib_transactions` and is mostly useful just for this study.

I absolutely do not want to claim `meta_build_date` because that's in the same namespace that the library "claims" (though claims is odd - more like, asks studies to populate and they will be understand in a certain way). Like, what if in a year, the library wants to use my `meta_build_date` table name but with a different schema.

So I was trying to find a reasonable table name that wouldn't conflict or that the library guaranteed that studies could have. I figured `meta` was good and `cumulus-library` wouldn't want it, since it likes specific names. But alternatively, I could avoid the `meta*` namespace altogether and put this metadata in `data_metrics__info` or something.

#### Tiny tables
Separately, if we're opening up the discussion about how things *should* work, I'd like to understand the value of different tables for each individual piece of metadata. What are the pros and cons of that vs just different columns in a single metadata table?

Reply:
OK I took the build date stuff out of this PR. This thread is useful and should continue, but is not necessarily relevant for this exact PR anymore.

Reply:
> I absolutely do not want to claim meta_build_date because that's in the same namespace that the library "claims" (though claims is odd - more like, asks studies to populate and they will be understand in a certain way). Like, what if in a year, the library wants to use my meta_build_date table name but with a different schema.

There's no postprocessing here - like, the library never does anything specific with meta tables. It's more of a 'by convention, if you have something that's not directly related to the clinical population, name it `study__meta_something` by convention, but nothing else is done with it unless the author chooses. so `study__meta` and `study__meta_build_date` have, to me, exactly the same right to exist. `meta` is not protected like `lib`, `etl`, or `nlp`, so it should be safe to call whatever - unless we :do: decide we want to use meta in this way, and then we can chat offline about what makes sense to exist in that way. We could have a group of supported metadata types, for example.

> Separately, if we're opening up the discussion about how things should work, I'd like to understand the value of different tables for each individual piece of metadata. What are the pros and cons of that vs just different columns in a single metadata table?

I think it's basically a 'do you want to create them as you go or all at once' kind of question. Tiny tables tend to point towards things like 'create a table after we've looked at a certain resource', where one big meta table would always be 'create this table at the end of your run'. I can see the appeal of both - so if you were thinking 'just grab a generic `study_meta` table and add additional fields to it later, i don't think that's wrong? We could be perscriptive here, but i don't think we need to, unless we wanted metadata to be more guided overall.

Reply:
I agree `meta` isn't protected like the others - but at the same time, the Library docs are pretty specific about what it likes to see in them, and how the fields should be interpreted.

If we're saying that we fully support studies going off script for that, we can never really have a separate tool read those tables. Like, arguably the aggregator shouldn't try to read `data_package_version`  (unless maybe it inspects the schema and it lines up with what we expect -- though even then, if we say the study author is under no obligation to take our guidance on how the field should be interpreted... :shrug:)

So it feels to me like the Library is definitely staking some claims on an unprotected namespace. Which maybe feels reasonable from the stance of "this namespace is *writable* for studies, unlike `lib__`, but it isn't fully owned by studies either" -- that's the position I thought we were in.

If we're instead in the territory of "we give advice for this namespace, but all bets are off"... maybe the docs should clarify that and we should re-consider how we consume the tables in any bits of the stack that do read them today.